File: app/src/main/java/com/mbmc/fiinfo/helper/CodeManager.java
package com.mbmc.fiinfo.helper;

import android.app.Activity;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;

import com.mbmc.fiinfo.R;
import com.mbmc.fiinfo.constant.Preferences;
import com.mbmc.fiinfo.ui.activity.MainActivity;


public class CodeManager {

    public static void send(Activity activity, int code) {
        if (!PreferencesManager.getInstance(activity)
                .getBoolean(Preferences.HIDE_CODE_INSTRUCTIONS)) {
            ((MainActivity) activity).showCodeInstructions(code);
        } else {
            openDialer(activity, code);
        }
    }

    public static void openDialer(Context context, int code) {
        copy(context, code);
        Intent intent = new Intent(Intent.ACTION_DIAL);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        context.startActivity(intent);
    }


    private static void copy(Context context, int code) {
        ClipboardManager clipboardManager =
                (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData clipData = ClipData.newPlainText(context.getString(R.string.code), context.getString(R.string.code_dialer, code));
        clipboardManager.setPrimaryClip(clipData);
    }

}

File: app/src/main/java/com/mbmc/fiinfo/helper/CodeManager.kt
package com.mbmc.fiinfo.helper

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Intent
import com.mbmc.fiinfo.R
import dagger.hilt.android.qualifiers.ApplicationContext
import javax.inject.Inject

class CodeManager @Inject constructor(
    @ApplicationContext private val context: Context,
    private val preferenceManager: PreferenceManager
) {

    fun execute(code: String): Boolean {
        if (preferenceManager.hasSeenCodeInstructions()) {
            openDialer(context, code)
            return true
        }
        return false
    }

    fun setHasSeenInstructions() {
        preferenceManager.setHasSeenCondeInstructions()
    }

    fun openDialer(context: Context, code: String) {
        copy(context, code)
        val intent = Intent(Intent.ACTION_DIAL).also {
            it.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        }
        context.startActivity(intent)
    }

    private fun copy(context: Context, code: String) {
        val clipboardManager = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        val clipData = ClipData.newPlainText(context.getString(R.string.code), context.getString(R.string.code_dialer, code))
        clipboardManager.setPrimaryClip(clipData)
    }
}